src/Microsoft.AspNet.Mvc.Core/ModelBinding/BodyModelBinder.cs

```diff
@@ -28,36 +27,45 @@ public BodyModelBinder()
         protected async override Task<ModelBindingResult> BindModelCoreAsync(
             [NotNull] ModelBindingContext bindingContext)
         {
+            // For compatibility with MVC 5.0 for top level object we want to consider an empty key instead of
+            // the parameter name/a custom name. In all other cases (like when binding body to a property) we
```

Comment:
Might need to file a bug for the explicit name (not the parameter name) case. Does it make sense to ignore the user's explicit name request?